c10/BUCK.oss

```diff
@@ -44,7 +45,12 @@ cxx_library(
     header_namespace = "c10",
     link_whole = True,
     platform_preprocessor_flags = [['windows', ['-D_WINDOWS', '-D_WIN32', '-DWIN32', '-DNOMINMAX', '-D_CRT_SECURE_NO_WARNINGS', '-D_USE_MATH_DEFINES']], ['windows.*64$', ['-D_WIN64']]],
-    preprocessor_flags = ['-DC10_BUILD_MAIN_LIB'],
+    preprocessor_flags = [
+        '-DC10_BUILD_MAIN_LIB',
+        '-DHAVE_MMAP=1',
```

Comment:
```
undefined symbol: _ZN3c1022RefcountedMapAllocator6decrefEv"
```

I had the same error in the bazel build when `HAVE_MMAP` wasn't being defined. Is it possible there is another build definition that needs the flag added?

If so, does that mean we have 4 completely unrelated build definitions: CMake, bazel, buck (oss), buck (internal)?

Reply:
@peterbell10 unfortunately it's even more than 4: more than 1 internal buck.

But they are not completely unrelated: all of them are supposed to use build.bzl files (like https://github.com/pytorch/pytorch/blob/main/c10/build.bzl), but it's only implemented for some things so far.

For this particular case, likely https://github.com/pytorch/pytorch/blob/main/c10/ovrsource_defs.bzl needs to be updated. But someone from Meta should work with you on this.